src/main/scala/ai/privado/languageEngine/java/passes/config/JavaPropertyLinkerPass.scala

```diff
@@ -70,17 +72,20 @@ class JavaPropertyLinkerPass(cpg: Cpg) extends PrivadoParallelCpgPass[JavaProper
 
     val annotatedMethodsList = annotatedMethods()
 
-    annotatedMethodsList
-      .filter { case (key, _) =>
-        propertyNode.name == Option(key.code.slice(3, key.code.length - 2)).getOrElse("")
-      }
-      .foreach { case (_, value) =>
-        builder.addEdge(propertyNode, value, EdgeTypes.IS_USED_AT)
-        builder.addEdge(value, propertyNode, EdgeTypes.ORIGINAL_PROPERTY)
+    annotatedMethods()
+      .filter { case (key, _) => propertyNode.name == Option(key.code.slice(3, key.code.length - 2)).getOrElse("") }
+      .foreach { case (_, method) =>
+        val referencedMember = method.ast.fieldAccess.referencedMember.l.headOption.orNull
```

Comment:
There can be multiple field access which may exist in a method, how can we guarantee the head is the one this property was used at?

Reply:
These methods are most commonly setters used to inject property values, and set them to a member. Almost all of them are one liners.

Reply:
Agree, but still I see some scope. Can you add a todo stating where this may not work, for future reference?

Reply:
@karan-batavia Ideally you should taint the method parameter. Which in turn will taint the right field access.

Reply:
@karan-batavia sorry for the confusion. What you are doing should work.